dnf-repoconfig-daemon/configuration.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * Licensed under the GNU Lesser General Public License Version 2.1
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "configuration.hpp"
+
+#include "../libdnf/dnf-context.h" // for find_base_arch
+#include "../libdnf/utils/os-release.hpp" // for getOsReleaseData
+
+#include <glob.h>
+#include <iostream>
+#include <rpm/rpmlib.h>
+
+void Configuration::read_configuration()
+{
+    set_substitutions();
+    read_main_config();
+    read_repo_configs();
+}
+
+void Configuration::set_substitutions()
+{
+    const char *arch;
+    rpmGetArchInfo(&arch, NULL);
+    substitutions["arch"] = std::string(arch);
+    const auto basearch = find_base_arch(arch);
+    substitutions["basearch"] = basearch ? std::string(basearch) : "";
+    //TODO do we have reliable way to detect releasever?
+    auto osdata = libdnf::getOsReleaseData();
```

Comment:
Maybe look at dnf_context_set_os_release(). It sets releasever by calling dnf_context_set_release_ver().